### PR TITLE
MAINT update dependencies for CivisML

### PR DESCRIPTION
**Pull request by elsander**
Updating `cloudpickle` and `joblib` versions following discussion in #329.

I manually tested this on Python 3.6.8 with up-to-date `cloudpickle` and `joblib` versions, and @jacksonllee did the same for Python 3.7.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,8 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 - Fixed issue with pyyaml version for Python 3.4 by requiring pyyaml version <=5.2
 
 ### Changed
+- Updated cloudpickle and joblib dependencies. (#349)
+- CivisML uses platform aliases instead of hard-coded template IDs. (#341)
```
Comment (jacksonllee) [inline CHANGELOG.md]: This entry duplicates the next line just below.

File: requirements.txt [hunk 2]
```diff
@@ -8,7 +8,7 @@ requests>=2.12.0,==2.*
 jsonschema>=2.5.1,<=3
 functools32>=3.2,<=3.99 ; python_version == '2.7'
 six>=1.10,<=1.99
-joblib>=0.11,<=0.13
+joblib>=0.11,<0.15
 pubnub>=4.0,<=4.99
 cloudpickle>=0.2,<=1 ; python_version != '3.4'
```
Comment (jacksonllee) [inline requirements.txt]: Just tested `pip install cloudpickle>=0.2,<=1` in a fresh virtual environment (Python 3.7, pip v19.3.1), and I got cloudpickle v1.0.0, which suggests the `<=1` seems like a bug if we meant to allow cloudpickle v1.2.2. How about `cloudpickle>=0.2,<2`? (I tested `pip install cloudpickle>=0.2,<2` and got cloudpickle v1.2.2 as desired.)

Also, apart from `requirements.txt` here, it's important (perhaps more so, for production purposes) to make the changes in `setup.py` as well.

Comment (elsander) [reply requirements.txt]: Yeah, I changed the wrong line :facepalm:. Good point about setup.py as well.

File: CHANGELOG.md (hunk 1 above)
Comment (elsander) [reply CHANGELOG.md]: merge conflicts FTW

File: setup.py [hunk 3]
```diff
@@ -58,9 +58,9 @@ def main():
             'requests>=2.12.0,==2.*',
             'jsonschema>=2.5.1,<=3',
             'six>=1.10,<=1.99',
-            'joblib>=0.11,<=0.13.99',
+            'joblib>=0.11,<=0.15',
```
Comment (jacksonllee) [inline setup.py]: `<0.15` instead of `<=0.15`?

File: requirements.txt [hunk 4]
```diff
@@ -8,7 +8,7 @@ requests>=2.12.0,==2.*
 jsonschema>=2.5.1,<=3
 functools32>=3.2,<=3.99 ; python_version == '2.7'
 six>=1.10,<=1.99
-joblib>=0.11,<=0.13
+joblib>=0.11,<0.15
 pubnub>=4.0,<=4.99
-cloudpickle>=0.2,<=1 ; python_version != '3.4'
+cloudpickle>=0.2,<=2 ; python_version != '3.4'
```
Comment (jacksonllee) [inline requirements.txt]: `<2` instead of `<=2`?

**Pull request merged**